uk.co.compucorp.civicrm.civihrcore/CRM/Smarty/plugins/function.civihrVersion.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2016                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright TTTP
+ * $Id$
+ *
+ */
+
+/**
+ * Display the CiviHR version
+ *
+ * @code
+ * The version is {crmVersion}.
+ *
+ * {crmVersion redact=auto assign=ver}The version is {$ver}.
+ * @endcode
+ *
+ * @param $params
+ * @param $smarty
+ *
+ * @return string
+ */
+function smarty_function_civihrVersion($params, &$smarty) {
+  $result = CRM_CiviHRCore::getVersion();
```

Comment:
You can use the ternary operator for that:

```
return CRM_CiviHRCore::getVersion() ?: t('unknown');
```

Note that this doesn't work very well for arrays because you would get an error if the index doesn't exist (different from Javascript where you would only get undefined). A better option is the beautiful null coalesce operator, which works both for methods and arrays, but it's only available in PHP 7: http://php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce